gcd/src/main/java/org/spine3/server/storage/datastore/DatastoreWrapper.java

```diff
@@ -20,220 +20,252 @@
-        final Entity.Builder entity = Entity.newBuilder()
-                .setKey(key)
-                .addProperty(makeProperty(VALUE_PROPERTY_NAME, makeValue(serializedMessage)));
-        return entity;
+    /*package*/ void delete(Key... keys) {
+        actor.delete(keys);
     }
 
     /**
-     * Converts the given {@link EntityResultOrBuilder} to the {@link Message}.
+     * Deletes all existing {@link Entities} of given kind.
      *
-     * @param entity  the entity to convert
-     * @param typeUrl the type url of the message
-     * @return the deserialized message
-     * @see Any#getTypeUrl()
+     * @param table Kind (a.k.a. type, table, etc.) of the records to delete.
      */
-    /* package */ static <M extends Message> M entityToMessage(@Nullable EntityResultOrBuilder entity, String typeUrl) {
+    /*package*/ void dropTable(String table) {
+        final Query query = Query.newEntityQueryBuilder()
+                .setKind(table)
+                .build();
+        final List<Entity> entities = read(query);
+        final Collection<Key> keys = Collections2.transform(entities, new Function<Entity, Key>() {
+            @Nullable
+            @Override
+            public Key apply(@Nullable Entity input) {
+                if (input == null) {
+                    return null;
+                }
 
-        if (entity == null) {
-            @SuppressWarnings("unchecked") // cast is safe because Any is Message
-            final M empty = (M) Any.getDefaultInstance();
-            return empty;
-        }
+                return input.getKey();
+            }
+        });
 
-        final Any.Builder any = Any.newBuilder();
+        final Key[] keysArray = new Key[keys.size()];
+        keys.toArray(keysArray);
 
-        final List<Property> properties = entity.getEntity().getPropertyList();
+        delete(keysArray);
+    }
 
-        for (Property property : properties) {
-            if (property.getName().equals(VALUE_PROPERTY_NAME)) {
-                any.setValue(property.getValue().getBlobValue());
-            }
-        }
+    /**
+     * Starts a transaction.
+     *
+     * <p>Since this method is called and until one of {@link #commitTransaction()} or {@link #rollbackTransaction()}
+     * is called, all CRUD operations on Datastore performed trough current instance of {@code DatastoreWrapper} become
+     * transactional.
+     *
+     * @throws IllegalStateException if a transaction is already started on this instance of {@code DatastoreWrapper}.
+     * @see #isTransactionActive()
+     */
+    @SuppressWarnings("WeakerAccess") // Part of API
+    /*package*/ void startTransaction() throws IllegalStateException {
+        checkState(!isTransactionActive(), NOT_ACTIVE_TRANSACTION_CONDITION_MESSAGE);
+        activeTransaction = datastore.newTransaction();
+        actor = activeTransaction;
+    }
+
+    /**
+     * Commits a transaction.
+     *
+     * <p>All transactional operations are being performed.
```

Comment:
`<p>Upon the method call, all the modifications within the active transaction are applied.`